### PR TITLE
Issue LIBCLOUD-417: Add support for more EC2 Elastic IP Address operations

### PR DESCRIPTION
**Pull request by oldpatricka**
Add support for:

DisassociateAddress
ReleaseAddress
AllocateAddress

Also fix AssociateAddress

### CODE REVIEW

File: libcloud/compute/drivers/ec2.py [hunk 1]
```diff
@@ -1155,6 +1155,34 @@ def _add_instance_filter(self, params, node):
             'Filter.0.Value.0': node.id
         })
 
+    def ex_allocate_address(self):
+        """
+        Allocate a new IP address
+
+        @rtype: C{str}
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: We've recently migrated to Sphinx style docstrings (yeah, still need to update documentation). Can you please update your code to use Sphinx style docstrings?

File: libcloud/compute/drivers/ec2.py [hunk 2]
```diff
@@ -1155,6 +1155,34 @@ def _add_instance_filter(self, params, node):
...
+        params = {'Action': 'AllocateAddress'}
+
+        response = self.connection.request(self.path, params=params).object
+        public_ip = findtext(element=response, xpath='publicIp',
+                                namespace=NAMESPACE)
+        return public_ip
+
+    def ex_release_address(self, elastic_ip_address):
+        """
+        Release an IP address
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: I would also update docstrings of all those new methods to clarify that those are elastic ip addresses (Release an elastic IP address and so on).

**Comment by Kami** [comment]
Can you please also add some examples of how to use this functionality to - https://github.com/apache/libcloud/blob/trunk/docs/compute/drivers/ec2.rst

And yes, the file is currently empty. The reason for that is that we just recently started bootstrapping provider specific documentation.

File: libcloud/compute/drivers/ec2.py (hunk 2 above)
Comment (oldpatricka) [inline libcloud/compute/drivers/ec2.py]: Good idea! I did it in a new commit (which I'll squash when this pr is ready)

**Comment by oldpatricka** [comment]
Ok, I switched to Sphinx style docstrings, and added an example to the docs. How does it look now?

**Comment by Kami** [comment]
@oldpatricka Looks good.

Please squash all the commits into one and attach patch to the corresponding JIRA ticket and I'll merge it into trunk.

**Comment by oldpatricka** [comment]
Great, the patch is attached to the ticket (https://issues.apache.org/jira/browse/LIBCLOUD-417), so I'm closing this request. Thanks for the prompt responses!

**Pull request closed without merging**